src/form-generator-multi-step.ts

```diff
@@ -83,30 +83,35 @@ export class FormGeneratorMultiStep {
         }
     }
 
-    generateForm = async (): Promise<HTMLDivElement> => {
-        const container = document.createElement('div');
-
-        for (let index = 0; index < formPages.length; index++) {
-            const page = formPages[index];
-            const formPage = await this.createPage(page.title, page.fields);
-            formPage.style.display = index === 0 ? 'block' : 'none';
-            container.appendChild(formPage);
-            (formPages[index] as any).element = formPage;
-        }
+    generateForm = async (): Promise<HTMLDivElement | undefined> => {
+        let container;
+        if (typeof document !== 'undefined') {
+            container = document.createElement('div');
+            for (let index = 0; index < formPages.length; index++) {
+                const page = formPages[index];
+                const formPage = await this.createPage(page.title, page.fields);
+                formPage.style.display = index === 0 ? 'block' : 'none';
+                container.appendChild(formPage);
+                (formPages[index] as any).element = formPage;
+            }
 
         document.body.appendChild(container);
 
         return container;
+        }
     }
+
 }
 
 const formGenerator = new FormGeneratorMultiStep(formConfig);
 const formValidator = new FormValidator();
 
 formGenerator.generateForm().then(form => {
-    document.body.appendChild(form);
-    form.addEventListener('submit', function (event) {
-        event.preventDefault();
-        formValidator.validation();
-    });
+    if (form) {
```

Comment:
czyli co ? jak forma nie ma to nie robimy nic ? :D pytanie kiedy go nie bedzie faktycznie ze ten if jest potrzebny

Reply:
@apawlik96 follow up do tego :D bo nie wiem w sumie co z tym ma byc

Reply:
warto zrobic jakies logowanie jak nie wejdzie w tego ifa zeby wiedziec co sie dzieje